website/oauth/models/__init__.py

```diff
@@ -394,7 +394,7 @@ class ApiOAuth2Application(StoredObject):
     callback_url = fields.StringField(required=True,
                                       validate=URLValidator())
 
-    def deactivate(self):
+    def deactivate(self, save=True):
```

Comment:
In general, we want to avoid implicit saves. Perhaps set the default to `False`?